bin/src/main.rs

```diff
@@ -396,6 +393,24 @@ async fn main() {
     }
 }
 
+fn load_agent_state(db_path: &DbPath) -> Result<Option<AgentState>, StateError> {
```

Comment:
Since it's returning AgentState/StateError it feels like this could be over in the state module, perhaps as a constructor on the AgentState type?

It's just a helper function though so I don't have a strong opinon.

Any thoughts?

Reply:
Yup, I was so focused on breaking out the functionality and testing it I just left it in the main file. Commit https://github.com/logdna/logdna-agent-v2/pull/226/commits/aa5313aa74261d210d5f24d1ff37dc19b0469a47 moves things around a bit.